fileio/ft_write_data.m

```diff
@@ -84,8 +84,11 @@ function ft_write_data(filename, dat, varargin)
 % determine the data size
 [nchans, nsamples] = size(dat);
 
-% ensure that the directory exists
-isdir_or_mkdir(fileparts(filename));
+% ensure that the directory exists (unless we're writing to a FieldTrip
+% buffer)
+if ~startsWith(filename, 'buffer:')
```

Comment:
rather than checking the start of the string, please use the `dataformat` variable